lib/semantics/scope.h

```diff
@@ -25,11 +25,16 @@
 #include <set>
 #include <string>
 
+namespace Fortran::fir {
+class ControlFlowAnalyzer;
+}
+
 namespace Fortran::semantics {
 
 using namespace parser::literals;
 
 class Scope {
+  friend fir::ControlFlowAnalyzer;
```

Comment:
Maybe `Scope`'s API can be extended to serve the needs of `ControlFlowAnalyzer` without needing to expose all of its content.

Reply:
Yet another TODO and rubbish from prototype hacking...